src/main/java/Deadline.java

```diff
@@ -0,0 +1,94 @@
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+import java.time.LocalDate;
+
+/**
+ * Represents a Deadline task. Represented by the task, the LocalDate object of the deadline,
+ * and a boolean state denoting whether the task has been done or not.
+ */
+
+public class Deadline extends Task {
+    private static final String OUTPUT_DATE_FORMAT = "MMM dd yyyy";
+    private final LocalDate deadline;
+
+    /**
+     * Constructor of a Deadline object
+     * @param task Task need to be done
+     * @param deadline Date when the task need to be done
+     * @throws DukeExceptionDeadline The date format is not in "yyyy-MM-dd"
+     */
+    Deadline(String task, String deadline) throws DukeExceptionDeadline {
+        super(task);
+        try {
+            DateTimeFormatter formatter = DateTimeFormatter.ISO_LOCAL_DATE;
+            this.deadline = LocalDate.parse(deadline, formatter);
+        } catch (DateTimeParseException e) {
+            System.out.println(e.getMessage());
+            throw new DukeExceptionDeadline("Wrong format of date." +
+                    " The format should be yyyy-MM-dd");
+        }
+    }
+
+    /**
+     * Constructor of a deadline object
+     * @param task Task need to be done
+     * @param deadline Date when the task need to be done
+     * @param done A boolean representing the state of the task (done or not)
+     * @throws DukeExceptionDeadline The date format is not in "yyyy-MM-dd"
+     */
+    Deadline(String task, String deadline, boolean done) throws DukeExceptionDeadline {
```

Comment:
Maybe you could consider naming the boolean to isDone such that it is more understandable?